File: src/EnchereService/RequestHelpers/MappingProfiles.cs
using AutoMapper;
using Contracts;
using EnchereService.DTOs;
using EnchereService.Models;

namespace EnchereService.RequestHelpers;

public class MappingProfiles : Profile
{
    public MappingProfiles()
    {
        CreateMap<Enchere, EnchereDto>().IncludeMembers(e => e.Item);
        CreateMap<Produit, EnchereDto>();
        CreateMap<CreateEnchereDto, Enchere>()
        .ForMember(d => d.Item, o => o.MapFrom(s => s));
        CreateMap<CreateEnchereDto, Produit>();
        CreateMap<EnchereDto, EnchereCreated>();
        CreateMap<Enchere, EnchereUpdated>().IncludeMembers(e => e.Item);
        CreateMap<Produit, EnchereUpdated>();
    }
}

File: src/RechercheService/services/requestHelpers/mappingProfiles.js
const automapper = require("automapper-ts");

automapper.initialize((config) => {
  config
    .createMap("EnchereCreated", "Produit")
    .forMember("_id", (opts) => opts.mapFrom("id"));

  config
    .createMap("EnchereUpdated", "Produit")
    .forMember("_id", (opts) => opts.mapFrom("id"));
});
